### PR TITLE
feat: support hide expandIcon

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [X] Bug fix
- [X] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

- 我们的 expandIcon 是 React.ReactNode 类型的，理论上应该支持传入 null 或 false 时隐藏展开按钮，但目前未隐藏

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | support hide expandIcon in menu  |
| 🇨🇳 Chinese | 菜单中支持隐藏展开图标  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at a8d25f1</samp>

This pull request adds a new feature to the `Menu` component, which allows users to customize or hide the icon that indicates a submenu. It also updates the dependency version of `rc-menu` and adds some tests for the new feature.

### <samp>🤖 Generated by Copilot at a8d25f1</samp>

* Add a new prop `expandIcon` to the `Menu` component to customize or hide the submenu icon ([link](https://github.com/ant-design/ant-design/pull/44554/files?diff=unified&w=0#diff-fb37d7abd9dddd9f2a8c076bfd9eea2336873f1051d1f37dceaca099206a4aedR486), [link](https://github.com/ant-design/ant-design/pull/44554/files?diff=unified&w=0#diff-bcc9a016f0b5e2844ec9d0aa399d06a86134f535931eeced33c3d08c3462ce43L129-R135))
* Update the dependency version of `rc-menu` to include the new feature ([link](https://github.com/ant-design/ant-design/pull/44554/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L137-R137))
* Remove unnecessary type assertions from the `useMemo` hook in the menu tests ([link](https://github.com/ant-design/ant-design/pull/44554/files?diff=unified&w=0#diff-1cad196e610d68ddf06e127c070d41cb9e4bfa00016e77c472841f261f154433L828-R828), [link](https://github.com/ant-design/ant-design/pull/44554/files?diff=unified&w=0#diff-1cad196e610d68ddf06e127c070d41cb9e4bfa00016e77c472841f261f154433L863-R863))
* Add a new test case to check the `expandIcon` prop behavior ([link](https://github.com/ant-design/ant-design/pull/44554/files?diff=unified&w=0#diff-1cad196e610d68ddf06e127c070d41cb9e4bfa00016e77c472841f261f154433R1126-R1156))

### CODE REVIEW

**Comment by kiner-tang** [comment]
> 原始问题是啥？

我们的 `expandIcon` 是 `React.ReactNode` 类型的，理论上应该支持传入 null 或 false 时隐藏展开按钮，但目前未隐藏

**Review by MadCcc — APPROVED** [review]

File: .dumi/pages/index/components/Theme/index.tsx [hunk 1]
```diff
@@ -483,6 +483,7 @@ export default function Theme() {
                 openKeys={['Design']}
                 style={{ height: '100%', borderRight: 0 }}
                 items={sideMenuItems}
+                expandIcon={false}
```
Comment (zombieJ) [inline .dumi/pages/index/components/Theme/index.tsx:486]: 这里为什么需要加 `expandIcon={false}` ?

Comment (kiner-tang) [reply .dumi/pages/index/components/Theme/index.tsx:486]: image
首页那边的菜单其实是受控的，无法折叠么？无法折叠的情况，感觉没有必要展示折叠按钮了

Comment (li-jia-nan) [reply .dumi/pages/index/components/Theme/index.tsx:486]: 我觉得还是加上比较好，不加的话感觉缺少一点东西

Comment (kiner-tang) [reply .dumi/pages/index/components/Theme/index.tsx:486]: 如果不能折叠还展示折叠按钮，很多人会以为是 bug，很多人不会想到是因为我们故意禁止折叠的

**Pull request merged**